### ISSUE TITLE
Effect of heat on stratification hindering convection/ DSW formation

### ISSUE THREAD

**Issue by willaguiar** [willaguiar/DSW-collaborative-project#40]
I was looking at profiles of $\sigma_0$ and $\delta \sigma_0 / \delta z$ for the *1mtop* and *5mtop* simulation, and noticed that at the upper 50 m, *1mtop* is less stratified than *5mtop* (below). 
 
<img width="901" alt="Screenshot 2024-06-26 at 5 04 22 PM" src="https://github.com/willaguiar/DSW-collaborative-project/assets/70033934/080cd9e0-4f10-4988-8906-77c86cfa643b">

Perhaps that stronger stratification in 5mtop requires more energy to break, hindering convection and DSW formation on 5mtop? 

That would add up to the fact that 5mtop has lower surface density, so it would have even lower potential energy to break stratification.

I cant explain why 5mtop would be more stratified tho.

**Comment by adele-morrison** [comment]
It would be interesting to see how these differences develop, eg what do these plots look like in the first month? Does the difference start from the surface and propagate downward?

**Comment by willaguiar** [comment]
Good point. I was just looking at that. Below are delta values (5m - 1m). each color is a month from IC, 1=Jan:
<img width="1056" alt="Screenshot 2024-06-26 at 7 01 18 PM" src="https://github.com/willaguiar/DSW-collaborative-project/assets/70033934/a121a5d8-749e-4962-8df3-e48a012b934c">

Looking at the $\Delta \sigma_0$.... perhaps a thinner cell is more efficiently stirred/mixed laterally by wind, making the surface denser. In turn, the thicker cell has move volume, is less efficiently mixed laterally by winds and some of its high density signal travels down to the bottom of the mixed layer ….. making the 5mtop more stratified?

**Comment by AndyHoggANU** [comment]
OK, so within the first month there is significantly lighter water in the upper 15m of the 5mtop case. And maybe more dense water in 15-50m range? That seems more like a difference in the vertical transport. I hate to say it, but a @PaulSpence -inspired salt budget over this first month (maybe at daily resolution?) might be informative??

**Comment by adele-morrison** [comment]
That's really interesting to see! It definitely starts at the surface in summer in the first month. It propagates down all of a sudden in May - I wonder if that's when some winter convection starts to happen normally in the control and in the 5mtop case the convection is weakened due to the already enhanced surface stratification.

Maybe going even closer to the start would be interesting, i.e. repeating the plot on the right of density profile anomalies using daily data for the first 10 days or so? I wonder if we'd see it starting constrained even more to the top layers?

**Comment by willaguiar** [comment]
Just updating here the same plots for panan 5mtop - 1mtop.  The behavior is fairly similar, but  perhaps more concentrated on the surface, and with a smaller intensity

<img width="930" alt="Screenshot 2024-06-28 at 3 15 43 PM" src="https://github.com/willaguiar/DSW-collaborative-project/assets/70033934/1844dcaa-cc16-4772-b812-2b8fd02c8efa">

**Comment by willaguiar** [comment]
Another plot here now, with the daily data for 5mtop - 1mtop, ACCESS-OM2. 
<img width="1257" alt="Screenshot 2024-06-30 at 12 44 13 PM" src="https://github.com/willaguiar/DSW-collaborative-project/assets/70033934/4bf41b59-8844-4d6b-b37b-7e4639579a1c">
We can see 5mtop progressively building up a lighter/fresher surface  and denser/saltier subsurface (~30m) in the first 29 days of simulation.

**Comment by willaguiar** [comment]
I was thinking....... 
As sea ice melts, it would add freshwater and volume on the surface cell of the model.  If I remember correctly from Andy 's $z*$ presentation, increases in *SSH* are distributed along $z*$ levels as changes in $dzt$, proportionally to the mean cell thickness. Perhaps that means that 5m cell can hold bigger $dzt$ increase and bigger increase in FW content than 1m cell.

Complementing that..... If I remember correctly, if the $z^*$ cell grows too much, then the extra grow relative to the maximum $z^{target}$ is used to calculate the vertical flux/velocity. I imagine it is easier to reach that maximum volume increase in the 1m cell, and thus its vertical velocities are likely bigger too ( exporting more of the extra FW to deeper levels and not a allowing to much FW to build up at surface).

This is a guess made knowing very little of the $z*$ vertical coords, so one question is....Does that make any sense?

**Comment by willaguiar** [comment]
Based on the density anomaly/stratification plots above, We thought it would be useful to see how the salinity over the shelf evolves in the few first months of the simulation + wether the freshwater transport to the shelf was altered between the simulations. 

For clarification, I calculated the freshwater transport as the residual of the mass transport (Qx, and Qy) minus the salt content , as in 
<img width="299" alt="Screenshot 2024-07-20 at 1 14 49 PM" src="https://github.com/user-attachments/assets/254b25e0-3404-413e-ac5d-5d10cf27e84a">
where S is the absolute salinity. And after that I extracted the freshwater transport across the 1km isobath.

First, we can see that there is a continuous decrease in salinity over the top 10 m meters of the shelf in the 5m model [c] in 90 days.
When we compute the freshwater transport across the 1km isobath, we can see below that right at the surface the freshwater transport of the 5m case increases a lot ([b] compared to [a]).  Even when integrated over an average Ekman Layer depth of 35 m, the 5m model has stronger southward freshwater transport [d].
<img width="917" alt="Screenshot 2024-07-20 at 1 49 28 PM" src="https://github.com/user-attachments/assets/1efd95eb-3076-4c46-a729-a2cc8de5b5b8">


This only clarifies that DSW shutdown is likely due to stronger Southward freshwater transport in the 5m. But that could still be either by  [differences in Ekman pumping/transport](https://github.com/willaguiar/DSW-collaborative-project/issues/41) or differences in ice melting I guess.

**Comment by willaguiar** [comment]
**Update on Freshwater/Salt effect- part 1**

Context: We know the MOM5_5m simulation has reduced DSW formation relative to MOM5_1m, because MOM5_5m shelf is fresher at surface, then resisting DSW formation by buoyancy fluxes. This is because MOM5_5m shelf gets very fresh/light at the surface in the first 30-60 days prior to form DSW (plot below). 
<img width="925" alt="Screenshot 2024-08-27 at 9 14 54 AM" src="https://github.com/user-attachments/assets/a194877f-3913-4499-ba95-24d25f0551a7">
The goal to evaluate the salt budget in the next comment, is to figure out what causes this freshening at the surface.

_____
**Salinity changes due to salt or freshwater content**
Salinity (S) can change either due to changing freshwater content, or due to changing salt content. With the DS/Dt output (salt tendency), I can calculate the daily salt content in each cell ($SC_{shelf}$). The daily mass (M) of each cell will be $M = SC_{shelf} \times$ $(S*1e-3)^{-1}$ , and the daily freshwater content of each cell will be the difference $FW_{shelf} = M - SC_{shelf}$  . With these values, I can calculate the salinity changes due to changes in salt content ($S^{salt}$) or changes in freshwater content ($S^{FW}$) as:
<img width="1218" alt="Screenshot 2024-08-27 at 9 29 07 AM" src="https://github.com/user-attachments/assets/5dd40546-2b29-4e94-b114-575f76618bfa">
A comparison of these three could give us a hint on whether salt or freshwater content changes determines the surface freshening in MOM5_5m. The biggest freshening on the shelf happens in the upper 10 m of the water column (below), so I focused on this layer.
<img width="908" alt="Screenshot 2024-08-21 at 9 18 36 AM" src="https://github.com/user-attachments/assets/5c3b6306-a7bd-4eb4-8d97-5c7aa1e4b1ac">
Below is $S^{salt}$ and $S^{FW}$ for the upper 10m, and $S^{salt}$ explains almost all the salinity changes. Therefore, the salt content budget might help us figure out the reason for the surface shelf freshening in MOM5_5m.
<img width="453" alt="Screenshot 2024-08-26 at 4 36 56 PM" src="https://github.com/user-attachments/assets/b567c74b-f5b3-4623-87f0-779f537c3d9f">

**Comment by willaguiar** [comment]
**Sal budget on the shelf**

For the salt budget I output the following terms, and summed them fluxes on the top 10m of the shelf:
<img width="548" alt="Screenshot 2024-08-26 at 5 26 17 PM" src="https://github.com/user-attachments/assets/5a93f738-43e1-4a83-b08f-631180435457">
With $F_{surf}$ being the sum of the outputs:  _'sfc_salt_flux_ice'_ + _'sfc_salt_flux_restore'_ 

Below is the salt budget terms for MOM5_1m (a,b) and MOM5_5m (b,c), and the 5m-1m difference (d,e) The dominant term in salt removal in MOM5_1m is the advection, which is approx. balanced by salt gain by the implicit diffusion (a,b). Both advective salt loss and the salt gain by implicit diffusion decreases in MOM5_5m (b,e), but the compensation is not perfect (perhaps due to increased stratification), resulting in stronger salt removal in the first 30 days.  
<img width="910" alt="Screenshot 2024-08-26 at 5 58 14 PM" src="https://github.com/user-attachments/assets/16eacf75-2ca3-4eec-bf2c-e457c1b4a067">
____
- Bias warning: I can’t seem to close the salt budget, i.e., DS/Dt do not match the right side of the equation. For context, below is Fsurf calculated from these flux diagnostics, and from the residual of the equation in the 1m case, and the residual is ~20 times smaller than the surface fluxes, so be mindful of that.But, the surface forcing difference between simulations still very small and and can’t explain salinity changes, independent if I compare the diagnostics or residual term. Meanwhile, any suggestion on how to close the salt budget is welcome.
<img width="689" alt="Screenshot 2024-08-26 at 7 20 10 PM" src="https://github.com/user-attachments/assets/2db43f39-ed27-4b10-aced-903d41cac0f3">

**Comment by adele-morrison** [comment]
For dS/dt are you using the online tendency term?

**Comment by willaguiar** [comment]
> For dS/dt are you using the online tendency term?

Yes -  for dS/dt I used the _salt_tendency_ output

**Comment by adele-morrison** [comment]
FYI, we solved the budget closure (`salt_vdiffuse_impl` already includes the surface fluxes).

**Comment by willaguiar** [comment]
Perhaps this plot below helps explain a little more our 5m signal…


Below are the change in SSH(a), Ekman pumping (b), vertical salt fluxes in the upper 20m (positive upwards, c,d), and salinity anomalies [3], all for the shelf only ( south of 1km isobath). All that for the absolute wind experiments. Blue/orange contours in d and e are 1m and 5m isopycnals respectively. 

Here a interesting story seem to appear. In the 5m case we have a fast development of positive anomalies in SSH near the Antarctic Coast (a, isobath shallower than 400m), suggesting increase in Ekman Pumping. The calculated Ekman pumping in the same region seem to increase from 1m to 5m cases.  The downwards salt advection also increases south of the 400m isobath (d,e) where the Ekman pumping increased, decreasing the salinity in the upper shelf, and making the shelf densities decrease (contours in d,e). 

This suggest that an increase in Ekman pumping, pumps out salt out of the surface, making these waters too fresh and DSW resistant.

![Ekman_pumping_explanation-2](https://github.com/user-attachments/assets/bbc4b3c9-2f5a-4d58-83ed-ea277f639b1e)

**Comment by willaguiar** [comment]
When we calculate the Ekman transport across the 1km isobath ( offline, assuming boussinesq, + southward), we see that the Ekman transport actually increases in the first month, which would increase the SSH at the coast, and increase the Ekman pumping… I think this ties up with our hypothesis than changes in Ekman pumping make the surface waters fresh and DSW resistant. The Ekman transport changes are small, but since we only change the top 5m of the grid, the pumping change likely happens only in this thin 5m slice ( ~15% of Ekman layer depth). So they might be important, even tho small.

<img width="895" alt="Screenshot 2024-09-02 at 12 10 59 PM" src="https://github.com/user-attachments/assets/e2949b4d-8d9e-4e59-96e6-53e2115135f4">

**Comment by adele-morrison** [comment]
That seems like it could be plausible.

But how can increasing thickness of the top cell increase the Ekman transport? Have we looked at maps of zonal and meridional velocity averaged over the top 5m? You say the speed is slower, but maybe by looking at zonal/meridional components, we could see if the direction has changed (e.g. slower but stronger meridional velocities perhaps could increase Ekman pumping at the coast).

**Comment by willaguiar** [comment]
Below are the mean velocities, average (weighted by dzt) over the top 5m of both simulations.
<img width="919" alt="Screenshot 2024-09-02 at 3 38 16 PM" src="https://github.com/user-attachments/assets/9eb49156-baf0-488d-aa70-09c37ff724a1">
In the zonal velocity (top) we clearly see the decrease, but in the meridional (bottom) is quite difficult to see, because is noisy. Perhaps seem to show the values shifting to be more negative towards the shelf.


I tried to simplify, and picked up the same velocities above, masked off-shelf, and averaged on the shelf (weighted by cell area).  Here it gets more clear... the zonal component does decrease, but the meridional component increases. 
<img width="923" alt="Screenshot 2024-09-02 at 3 41 45 PM" src="https://github.com/user-attachments/assets/6f18926b-7c7b-4821-ba34-62980314ddd2">

So yepp, even tho the model with thicker cell has slower speeds, the meridional component of the velocity is bigger, increasing the Southward Ekman transport and pumping near the coast.

**Comment by adele-morrison** [comment]
That seems quite a significant change. Why would the meridional velocities increase as vertical resolution decreases? Could it be related to the direction of the Ekman transport rotating? What’s the average angle / direction of the transport in the top 5m on the shelf?

**Comment by willaguiar** [comment]
Yes, I think it is due to the change in the angle of the transport/velocities in the top 5m. 

Below are the average 5m speed angle in both simulations, on the shelf. 0 degrees is northward speed, and the angle increases clockwise. That is not the same as the Ekman transport, but it could give a hint. (I'll calculate the Ekman transport angle later too). So the mean speed is directed Southwest. In the 5m case, the angle decreases by ~2 degrees, which would turn velocities a little more towards the south. This is simple to see on the shelf averages, but harder on the shelf maps.
<img width="925" alt="Screenshot 2024-09-02 at 4 35 32 PM" src="https://github.com/user-attachments/assets/b496a035-9f6d-4ba9-950d-da8a449db8a4">

**Comment by willaguiar** [comment]
To get a better sense on if/why the Ekman transport really changes with resolution, I ran a new set of single column experiments. They are initialized with temperature and salinity as in [these previous experiments for the Weddell Sea]( https://github.com/willaguiar/DSW-collaborative-project/issues/34#issuecomment-1935342666) , use a beta plane approximation, with mean f0 for $65 ^oS$. Experiments have 2 zonal cells and 2 meridional cells with dx=dy=0.1 degrees, a maximum depth of 500 m and are re-entrant in X and Y. The only forcing applied is a zonal wind stress Tx = -0.05 n/m2 to simulate a westward wind. The 1m and 5m vertical grid are the same as MOM5_1m and MOM5_5m, but for the upper 500m of the water column.

____
Mean daily velocities after 30 days
Just by changing the grid the velocity profiles change. 
<img width="916" alt="Screenshot 2024-09-11 at 3 46 24 PM" src="https://github.com/user-attachments/assets/f6245a7d-195b-403d-bea6-fa1524c471f0">

We also see an increase in the Southward meridional transport in the 5m case, and decrease in the westward zonal transport (perhaps quite small??)
<img width="906" alt="Screenshot 2024-09-11 at 3 47 15 PM" src="https://github.com/user-attachments/assets/5c191bf8-65a0-43dc-918a-24ae4820424c">

Per layer, there is a decrease in meridional transport only in the top 5m of the “MOM5_5m” simulation, but right below it the southward transport increases (hence total increase along the water column in the bar plot)
<img width="917" alt="Screenshot 2024-09-11 at 3 51 15 PM" src="https://github.com/user-attachments/assets/ca91a47d-8b22-4ccb-95f1-6461f0bcb3ea">

And below is a visualization of the direction of the velocities. The top 4 cells (total dzt~5m) of 1mtop case were weighted-average for comparison with the first cell of the 5m case. The results show that the surface speeds are tilted southeast in the 5m, while the subsurface layers are tilted more to the west in the 5m case.
<img width="884" alt="Screenshot 2024-09-11 at 4 54 08 PM" src="https://github.com/user-attachments/assets/cb494bcb-795e-4ba2-b52a-f69e98a7e410">
So the single column experiments agree that coarsening the grid changes the meridional mass transport, and the direction of the transport, as we noticed for access-om2 MOM5_5m.
I guess the link missing is …. To come up with a way to calculate how much the change in transport direction/meridional transport value would alter the SSH and Ekman pumping?

**Comment by willaguiar** [comment]
I reproduced the plots above in a rerun of the 1mtop, in which I used the KDS75 grid (1mtop) for running, but saving the output in the 5mtop grid. The resulting velocities seem to shift the same way in the online-depth-averaged output. 
<img width="903" alt="Screenshot 2024-09-16 at 10 06 36 AM" src="https://github.com/user-attachments/assets/ff87d133-814f-4a93-91ee-3ad723a99b80">
<img width="916" alt="Screenshot 2024-09-16 at 10 06 44 AM" src="https://github.com/user-attachments/assets/d5095f8d-5c62-45df-8a03-bbdd7c9988b0">
<img width="927" alt="Screenshot 2024-09-16 at 10 06 53 AM" src="https://github.com/user-attachments/assets/77f0755f-3942-4fb4-97f5-dac381b4e89c">
<img width="879" alt="Screenshot 2024-09-16 at 10 09 41 AM" src="https://github.com/user-attachments/assets/26f92723-4cac-49c9-aaf3-0710ac90c54c">



I wonder if it is worthy reruning (~1 year) of panan_1mtop with the same approach of saving the outputs in 5m grid?

**Comment by adele-morrison** [comment]
I don't understand. Is that just saying that it makes no difference if you depth average the output of top 5m of 1m_top offline or online (because it shouldn't)?

**Comment by willaguiar** [comment]
yes... that was just a test to see if averaging the output online would give different results than offline ( it doesn't)

**Comment by adele-morrison** [comment]
Oh sorry, what I meant above was not averaging the output, but actually averaging the velocities used by the model every timestep online.

**Comment by wghuneke** [comment]
> I think the best way to prove that the change in velocities drive the freshwater change is what Bob and Alistair suggested: Homogenise (i.e. depth average) the horizontal velocities in the top 5m of the 1m simulation online. Maybe worth testing offline first how well the depth average of the 1m case matches the 5m.

Would be neat if we could show this!

**Comment by willaguiar** [comment]
I was looking at MOM6 options, and I noticed it had an option to apply the wind stress not only in the first cell, but within a depth range (`DIRECT_STRESS = True`). If we achieve similar decrease in DSW formation in a simulation with the winds stress spread over the top 5m (`HMIX_STRESS=5.1 m`), but still having a 1m grid vertical, then that would be a good indication that a thicker upper cell feels the winds/Ekman transports differently, creating the pocket of low salinity waters we saw in [this comment]( https://github.com/willaguiar/DSW-collaborative-project/issues/40#issuecomment-2323641563)

Below is the result of this run (green lines - `Panan_1m_direct `experiment). We can see that ~100% of SWMT changes with resolution, and ~70% of the DSW overflow changes with resolution are reproduced by the `Panan_1m_direct` experiment, agreeing with our hypothesis.
<img width="782" alt="Screenshot 2024-09-30 at 9 30 46 AM" src="https://github.com/user-attachments/assets/0bac52f7-4eb5-4368-90b9-0128ec4fcbc2">
Finally, when looking at the signals that are relevant for Ekman dynamics (below for first Feb monthly mean), we can see that `Panan_1m_direct `has [a] an increase in SSH close the shelf similar to `Panan_5m`, similar mass transport across the 1km isobath at surface as `panan_5m` [b,c]. [d] is the salinity anomaly (`panan_5m` – `panan_1m`), and [e] is the anomaly relative to the direct winds case ( `panan_5m` – `panan_1m_direct`). We can see that most of the surface salinity anomaly also disappears in [e].
<img width="1165" alt="Screenshot 2024-09-30 at 12 11 03 PM" src="https://github.com/user-attachments/assets/84a64ab6-ae05-4c94-bf20-0e38858a07b6">
**Bottomline:** A thicker surface cell has increased Southward Ekman transport and coastal pumping (e.g., SSH), [exporting more salt out of the surface (panel d)]( https://github.com/willaguiar/DSW-collaborative-project/issues/40#issuecomment-2362880686) and making the surface fresh and DSW resistant.


What do you think of that as the final explanation for the MS? What are your thoughts/suggestions on it?

@AndyHoggANU @adele-morrison @wghuneke @PaulSpence @fabiobdias @dkhutch

**Comment by AndyHoggANU** [comment]
Hi @willaguiar - nice work! Yes, I agree that this `DIRECT_STRESS` option is a very nice test of exactly what we wanted to test. It provides pretty firm evidence that the vertical resolution of the Ekman spiral is responsible for the differences we see.

It remains a very surprising result — that the system ask whole is so sensitive to the direct of the near-surface velocities — but the evidence supporting it is now very strong.

The one thing I would query is whether or not we can so that it is coastal pumping that is important, or if it’s cross-shelf freshwater transport. I don’t think we can yet distinguish … but perhaps we don’t need to resolve that question, and can leave it slightly open??

**Comment by willaguiar** [comment]
Thanks @AndyHoggANU 

I have one extra question...I have been trying to calculate the vertical salt flux, in something like this plot below but for MOM6. However, MOM6 don't have any option for vertical flux diagnostics ( neither velocity, mass flux, nor salt flux). So my approach have been to calculate vertical mass fluxes based on the daily mass convergence in each cell (kg/s). then I do an upwards cumulative sum of the mass convergence in kg/s, to get the vertical mass fluxes. Finally I multiply these vertical mass fluxes (kg/s) by the absolute salinity times 10^-3 to get the vertical salt flux.

I just wanted to gather an opinion on ... does that approach make sense?  Is there other way of calculating offline vertical mass/salt fluxes ? I understand that in z* cells grown and shrink, do I have to take that into account together with the changes in densities in each cell?
<img width="543" alt="Screenshot 2024-10-16 at 4 45 55 PM" src="https://github.com/user-attachments/assets/dbb2806b-7069-4c1a-a17e-0c0409cb2562">

**Comment by willaguiar** [comment]
Hello Everyone

I have finished editing the paper figures according to our last meeting suggestions, and should have a version of the paper for internal review by next week. Meanwhile, I updated the [figure outline ](https://github.com/willaguiar/DSW-collaborative-project/blob/main/Figures_outline.md) file... could you all check it out and let me know in case you have further suggestions or ideas? 
@AndyHoggANU @adele-morrison @wghuneke @PaulSpence @dkhutch @fabiobdias 

PS: I am also inviting Kial to be co-author in the paper.
